backend/de.metas.swat/de.metas.swat.webui/src/main/java/de/metas/archive/process/MassConcatenateOutboundPdfs.java

```diff
@@ -93,13 +98,36 @@ protected String doIt() throws Exception
 		return "OK/Error # " + printedIds.size() + "/" + errorCount.get();
 	}
 
-	private Stream<AdArchive> streamArchivesForFilter(@NonNull final IQueryFilter<I_C_Doc_Outbound_Log> outboundLogFilter)
+	@NonNull
+	private Stream<AdArchive> streamArchivesForFilter()
 	{
-		return queryBL.createQueryBuilder(I_C_Doc_Outbound_Log.class)
-				.addOnlyActiveRecordsFilter()
-				.filter(outboundLogFilter)
+		final IQueryBuilder<I_C_Doc_Outbound_Log> queryBuilder = queryBL.createQueryBuilder(I_C_Doc_Outbound_Log.class)
+				.addOnlyActiveRecordsFilter();
+
+		final ImmutableList<DocOutboundLogId> ids;
+		if (getSelectedRowIds().isAll())
+		{
+			ids = getView().getLastOrderedSelectionIds(0, ALL_ITEMS_LIMIT)
+					.stream()
+					.map(id -> id.toId(DocOutboundLogId::ofRepoId))
+					.collect(ImmutableList.toImmutableList());
+		}
+		else
+		{
+			ids = getSelectedRowIds().toListIds(DocOutboundLogId::ofRepoId);
+		}
+
+		final Map<DocOutboundLogId, Integer> orderMap = new HashMap<>();
+		for (int i = 0; i < ids.size(); i++)
+		{
+			orderMap.put(ids.get(i), i);
+		}
+
+		return queryBuilder
+				.addInArrayFilter(I_C_Doc_Outbound_Log.COLUMNNAME_C_Doc_Outbound_Log_ID, ids)
 				.create()
 				.iterateAndStream()
+				.sorted(Comparator.comparingInt(o -> orderMap.getOrDefault(DocOutboundLogId.ofRepoId(o.getC_Doc_Outbound_Log_ID()), Integer.MAX_VALUE)))
```

Comment:
this kills any streaming performance because we have to accumulate everything in memory in order to make sure it's "sorted"